tests/scripts/all.sh

```diff
@@ -873,7 +873,12 @@ component_check_doxygen_warnings () {
     tests/scripts/doxygen.sh
 }
 
+component_check_test_requires_psa_disabled () {
+  msg "Check: tests requiring PSA to be disabled"
 
+  not grep -n -R 'depends.*!MBEDTLS_USE_PSA_CRYPTO' tests/suites/
+  not grep -n -R 'requires.*disabled.*USE_PSA' tests/ssl-opt.sh tests/opt-testcases/
```

Comment:
Should we really have these policies? You managed to get rid of all the non-usepsa tests, but it's not implausible that at some point we'll need to add tests that have distinct versions in the PSA and non-PSA case. For example tests that must use a specific RNG seed, which is done completely differently through mbedtls and psa APIs.

Reply:
I don't have a stong opinion about theses policies, I'm ok to remove it since it's not part of the original work anyway.